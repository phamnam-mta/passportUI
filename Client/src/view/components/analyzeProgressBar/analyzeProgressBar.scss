@import "~@fluentui/react/dist/sass/References";

.analyze-progress-bar {
    display: flex;
    justify-content: center;
    flex-direction: column;
    flex-grow: 1;
    width: 30%;
    gap: 10px;
    opacity: 1;

    .progress-details {
        display: flex;
        flex-direction: column;
        gap: 5px;

        .progress-title {
            font-size: 16px;
            font-weight: 600;
        }

        .progress-subtitle {
            white-space: nowrap;
            overflow: hidden;
            text-overflow: ellipsis;
            font-size: 12px;
            color: $ms-color-gray120;
        }
    }
}
