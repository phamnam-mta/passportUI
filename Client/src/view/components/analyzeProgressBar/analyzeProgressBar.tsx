import * as React from "react";
import { ProgressIndicator } from "@fluentui/react";

import "./analyzeProgressBar.scss";

interface IAnalyzeProgressBarProps {
    title: string;
    subtitle: string;
    percentComplete: number | undefined;
}

export class AnalyzeProgressBar extends React.PureComponent<IAnalyzeProgressBarProps> {
    public render() {
        const { title, subtitle, percentComplete } = this.props;
        return (
            <div className="analyze-progress-bar">
                <section className="progress-details">
                    <article className="progress-title">{title}</article>
                    <article className="progress-subtitle">{subtitle}</article>
                </section>
                <ProgressIndicator percentComplete={percentComplete} />
            </div>
        );
    }
}
