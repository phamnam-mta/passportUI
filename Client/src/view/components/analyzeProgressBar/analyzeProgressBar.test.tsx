import * as React from "react";
import { shallow } from "enzyme";

import { AnalyzeProgressBar } from "./analyzeProgressBar";

describe("<AnalyzeProgressBar />", () => {
    let baseProps;

    beforeEach(() => {
        baseProps = {
            title: "<title>",
            subtitle: "<subtitle>",
            percentComplete: undefined,
        };
    });

    describe("Rendering", () => {
        it("should render correctly", () => {
            const wrapper = shallow(<AnalyzeProgressBar {...baseProps} />);

            expect(wrapper).toMatchSnapshot();
        });
    });
});
