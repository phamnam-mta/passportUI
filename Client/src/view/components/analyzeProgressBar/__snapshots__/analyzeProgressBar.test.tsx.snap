// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AnalyzeProgressBar /> Rendering should render correctly 1`] = `
<div
  className="analyze-progress-bar"
>
  <section
    className="progress-details"
  >
    <article
      className="progress-title"
    >
      &lt;title&gt;
    </article>
    <article
      className="progress-subtitle"
    >
      &lt;subtitle&gt;
    </article>
  </section>
  <StyledProgressIndicatorBase />
</div>
`;
